src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -912,7 +935,18 @@ public async ValueTask DisposeAsync()
             // ended by now
             _cancel.Cancel();
             if (_connectMembers != null)
-                await _connectMembers.CfAwaitCanceled();
+            {
+                try
+                {
+                    await _connectMembers.CfAwait(120_000); // give it 2 mins
```

Comment:
What could be result if we don't wait here since disposing is happening? Is there a reason to wait?